patches/tModLoader/Terraria/ModLoader/UI/UIModSourceItem.cs

```diff
@@ -205,17 +205,23 @@ protected override void DrawSelf(SpriteBatch spriteBatch)
 
 		// Display Run tModPorter when .csproj is valid
 		if (sourceUpgradeTask is { IsCompleted: true }) {
-			bool result = sourceUpgradeTask.Result;
+			try {
+				bool result = sourceUpgradeTask.Result;
```

Comment:
Recommended to change this to `.GetAwaiter().GetResult()` to avoid wrapping the exception in an `AggregateException` (yeah, .NET didn't do great here)